src/dvc_data/index.py

```diff
@@ -27,6 +27,24 @@ class DataIndexEntry:
 
     loaded: Optional[bool] = None
 
+    def as_dvc_info(self):
+        """
+        Convert self to a dvc_info dict.
+
+        Note that the dict does NOT contain a "name" key.
+        """
+        isdir = self.hash_info and self.hash_info.isdir
+        return {
+            "type": "directory" if isdir else "file",
+            "size": self.meta.size if self.meta else 0,
+            "isexec": self.meta.isexec if self.meta else False,
+            "isdvc": True,
+            "isout": True,
+            "obj": self.obj,
+            "entry": self,
+            self.hash_info.name: self.hash_info.value,
+        }
```

Comment:
Do you think this will come in handy later? Just seems like it might be enough to keep it in `info` with the rest of the logic and not spread it to DataIndexEntry, where we don't yet have any use for it (dvc_info name is also a bit confusing in this particular context).

Reply:
I thought it would be helpful to have it for `.ls()` but I ended up not using it. I feel that it could make sense, but we don't need it for now.